### PR TITLE
 Implement custom project support in Kitura CLI

### PR DESCRIPTION
**Pull request by ShihabMehboob**

### CODE REVIEW

File: .swift-version [hunk 1]
```diff
@@ -0,0 +1 @@
+4.1
```
Comment (ianpartridge) [inline .swift-version]: This is an NPM project so it shouldn't have a `.swift-version` file.

File: kitura-init.js [hunk 2]
```diff
@@ -34,8 +34,13 @@ validateDirectoryName();
 // Make sure directory is empty.
 checkCurrentDirIsEmpty();
 
-// Clone repo contents into current directory.
-cloneProject(initURL, initBranch);
+// If '--clone' specified, then clone project from provided git repository.
+if ((args.includes('--clone'))) {
+  cloneProject(process.argv[3], initBranch);
```
Comment (ianpartridge) [inline kitura-init.js]: This means we'll always look for an `init` branch, which doesn't seem right. Would it be better to do it like this:

```
kitura init --repo ianpartridge/myproject --branch mybranch
```

So we automatically add `github.com` etc. if it's not provided, and let people specify a branch if they want?  If they don't add `--branch` we use `master` by default.

File: package-lock.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "kitura-cli",
-  "version": "0.0.12",
+  "version": "0.0.13",
```
Comment (ianpartridge) [inline package-lock.json]: Versioning should happen as a separate PR once the feature is merged, please.

File: test.sh [hunk 4]
```diff
@@ -52,6 +52,19 @@ fi
 cd ..
 rm -rf $DIRNAME
 
+echo "Testing: kitura init --clone https://github.com/IBM-Swift/generator-swiftserver-projects"
```
Comment (ianpartridge) [inline test.sh:55]: Thanks for adding the test!

File: test.sh [hunk 5]
```diff
@@ -52,6 +52,19 @@ fi
...
+mkdir $DIRNAME
+cd $DIRNAME || exit 1
+if ! kitura init --clone https://github.com/IBM-Swift/generator-swiftserver-projects
+then
+echo "Failed"
```
Comment (ianpartridge) [inline test.sh]: Please indent this like elsewhere in the file.

File: kitura-init.js (hunk 2 above)
Comment (ShihabMehboob) [reply kitura-init.js]: Amended it to default to the master branch, and accept a --branch param.

**Comment by djones6** [comment]
Closing as this would require re-implementing on top of the Go version (#62).

**Pull request closed without merging**